lib/formtastic/inputs/check_boxes_input.rb

```diff
@@ -180,11 +180,18 @@ def checkbox_input(choice)
       def make_selected_values
         if object.respond_to?(method)
           selected_items = object.send(method)
-
           # Construct an array from the return value, regardless of the return type
           selected_items = [*selected_items].compact.flatten
 
-          [*selected_items.map { |o| send_or_call_or_object(value_method, o) }].compact
+          selected = []
+          selected_items.each do |o|
+            item = send_or_call_or_object(value_method, o)
+            if item == nil
+              item = o.id if o.respond_to? :id
+            end
+            selected << item
+          end
+          selected
```

Comment:
Syntax: this could be written as

```ruby
selected_items.map do |selected_item|
  selected_item_id = selected_item.id if selected_item.respond_to?(:id)
  send_or_call_or_object(value_method, selected_item) || selected_item_id
end
```